src/lsm/manifest_log.zig

```diff
@@ -255,14 +309,29 @@ pub fn ManifestLogType(comptime Storage: type) type {
                 const table = &tables_used[entry];
                 assert(table.tree_id >= manifest_log.options.tree_id_min);
                 assert(table.tree_id <= manifest_log.options.tree_id_max);
+                assert(table.address > 0);
 
-                if (manifest.insert_table_extent(table.address, block_reference.address, entry)) {
-                    switch (label.event) {
-                        .insert => manifest_log.open_event(manifest_log, label.level, table),
-                        .remove => manifest.queue_for_compaction(block_reference.address),
-                    }
+                if (label.event == .remove) {
+                    assert(manifest_log.tables_removed.get(table.address) == null);
+
+                    manifest.queue_for_compaction(block_address);
+                    manifest_log.tables_removed.putAssumeCapacity(table.address, {});
                 } else {
-                    manifest.queue_for_compaction(block_reference.address);
+                    if (manifest_log.tables_removed.get(table.address)) |_| {
+                        manifest.queue_for_compaction(block_address);
+
+                        if (label.event == .insert) {
+                            assert(manifest_log.tables_removed.remove(table.address));
+                        }
+                    } else {
+                        var extent = manifest_log.tables.getOrPutAssumeCapacity(table.address);
+                        if (extent.found_existing) {
+                            manifest.queue_for_compaction(block_address);
```

Comment:
`assert(label.event == .update)`

Reply:
I don't think this holds -- the event that created the existing extent was an `.update`, but the one encountered now precedes it chronologically, so it may be an `insert`.